### PR TITLE
Making block impl tags unique

### PR DESCRIPTION
**Pull request by john-science**
## What is the change?

This PR makes all the sphinx-needs `impl` tags in the repo unique.

## Why is the change being made?

We had some duplicate `impl` tags in `blocks.py`.

---

## Checklist

- [X] This PR has only [one purpose or idea](https://terrapower.github.io/armi/developer/tooling.html#one-idea-one-pr).
- [X] [Tests](https://terrapower.github.io/armi/developer/tooling.html#test-it) have been added/updated to verify that the new/changed code works.

- [X] The code style follows [good practices](https://terrapower.github.io/armi/developer/standards_and_practices.html).
- [X] The commit message(s) follow [good practices](https://terrapower.github.io/armi/developer/tooling.html).

- [X] The [release notes](https://terrapower.github.io/armi/release/index.html) (location `doc/release/0.X.rst`) are up-to-date with any important changes.
- [X] The [documentation](https://terrapower.github.io/armi/developer/tooling.html#document-it) is still up-to-date in the `doc` folder.
- [X] No [requirements](https://terrapower.github.io/armi/developer/tooling.html#watch-for-requirements) were altered.
- [X] The dependencies are still up-to-date in `pyproject.toml`.

### CODE REVIEW

**Comment by john-science** [comment]
@bdlafleur Just FYI.

File: armi/nuclearDataIO/tests/test_xsCollections.py [hunk 1]
```diff
@@ -138,6 +138,13 @@ def r(self, r):
         self._r = r
 
     def getVolume(self, *args, **kwargs):
+        """
+        Return the volume of a block.
+
+        .. impl:: Volume of block is retrievable
+            :id: I_ARMI_BLOCK_DIMS10
+            :implements: R_ARMI_BLOCK_DIMS
+        """
```
Comment (opotowsky) [inline armi/nuclearDataIO/tests/test_xsCollections.py:147]: Is this the right spot for the `getVolume` impl tag? I thought the target impl was in blocks.py.

Comment (john-science) [reply armi/nuclearDataIO/tests/test_xsCollections.py]: Whoops.  Fixed!

**Review by opotowsky — APPROVED** [review]
Such a good find! Note that the missing docstring issue may pop up with internal repos as well, but I think you have that covered already

Seems like a scan of missing docstrings is in order 🤖

**Pull request merged**